### PR TITLE
Port MIRROR_PAD kernel from TFL

### PR DESCRIPTION
**Pull request by mansnils**
This PR contains several commits for readable history and ease of review.

This is PR3 in the sequence of PRs, outlined in issue:
BUG=Fixes https://github.com/tensorflow/tflite-micro/issues/741

### CODE REVIEW

File: tensorflow/lite/micro/kernels/mirror_pad.cc [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+                 int* output_dims_num_elements, int* input_dims_num_elements,
+                 const int offset) {
+  int flat_index = 0;
+  int64_t left_pad = 0, right_pad = 0, dimension_index, index_in_input;
+
+  for (int i = 0; i < num_dims; ++i) {
+    switch (padding_matrix->type) {
+      case kTfLiteInt32:
+        GetPadding(padding_matrix->data.i32, i, &left_pad, &right_pad);
+        // printf("correct  %d %d\n", left_pad, right_pad);
```
Comment (njeffrie) [inline tensorflow/lite/micro/kernels/mirror_pad.cc:71]: remove

**Review by njeffrie — APPROVED** [review]
Looks good other than the one dead printf

File: tensorflow/lite/micro/kernels/mirror_pad.cc (hunk 1 above)
Comment (mansnils) [reply tensorflow/lite/micro/kernels/mirror_pad.cc]: done

**Comment by mansnils** [comment]
@njeffrie ping for review

**Comment by mansnils** [comment]
@njeffrie @advaitjain all comments should be resolved

**Review by njeffrie — APPROVED** [review]

**Pull request merged**